src/Symfony/Component/DependencyInjection/Tests/Compiler/ResolveClassPassTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Tests\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\ResolveClassPass;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Tests\Fixtures\CaseSensitiveClass;
+
+class ResolveClassPassTest extends \PHPUnit_Framework_TestCase
+{
+    public function provideValidClassId()
+    {
+        yield array('Acme\UnknownClass');
+        yield array(CaseSensitiveClass::class);
+    }
+
+    /**
+     * @dataProvider provideValidClassId
+     */
+    public function testResolveClassFromId($serviceId)
+    {
+        $pass = new ResolveClassPass();
+        $container = new ContainerBuilder();
+        $def = $container->register($serviceId);
+
+        $pass->process($container);
+
+        $this->assertSame($serviceId, $def->getClass());
+    }
+
+    public function provideInvalidClassId()
+    {
+        yield array(\stdClass::class);
+        yield array('bar');
+        yield array('\DateTime');
+    }
+
+    /**
+     * @dataProvider provideInvalidClassId
+     */
+    public function testWontResolveClassFromId($serviceId)
```

Comment:
It's a personal preference. I prefer showing the dataset before the testcase itself, as I find it more readable like this (the test case might be long, so having the dataset right before the method allows to quickly see the tested data, whereas having it after the method may not fit your screen).